### PR TITLE
Contributes script to install IDAES and COIN solvers on Google Colab

### PR DESCRIPTION
**Pull request by adowling2**
Contributes script to install IDAES and COIN solvers on Google Colab

## Summary/Motivation:
- Installing IDAES (and solvers via get-extensions) is a little quirky on Colab
- We've used (a prior version of) this script for teaching with Google Colab for over a year with >40 student testers

## Changes proposed in this PR:
- Contributes one file

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by adowling2** [comment]
@jsiirola is another good reviewer.

I'd like feedback on:
- How should this get tested?
- Are we okay with the API / level of commenting?
- Are we okay with leaving the `install_CBC`, etc. backup methods? The default usage is just to run `install_idaes()` and then `install_ipopt()`

**Review by jsiirola — CHANGES_REQUESTED** [review]
In addition to the specific comments below, there are several general changes that should be made:
- fix the identified spelling errors
- remove the logic for direct downloads from `ampl.com`
- I would suggest replacing (all?) uses of `os.system` with `subprocess.run`

File: scripts/colab_helper.py
Comment (adowling2) [reply scripts/colab_helper.py:126]: I just removed these.

File: scripts/colab_helper.py
Comment (adowling2) [reply scripts/colab_helper.py]: Thanks for the great suggestion. I implemented it.

File: scripts/colab_helper.py
Comment (adowling2) [reply scripts/colab_helper.py]: Also, I want to make these solvers accessible without `import idaes` for two reasons:
* Have the script print the solver versions, which is a good check and helpful for debugging.
* Reuse this script in workshops/classes/tutorials that introduce Pyomo. In my experience, skipping `import idaes` is easier with brand-new Pyomo users.

**Comment by adowling2** [comment]
@jsiirola This is ready for you to review.

@dangunter You might also find this helpful/interesting. I've been using a script like this to install Pyomo and IDAES in Google Colab for my graduate elective.

**Comment by adowling2** [comment]
@ksbeattie @lbianchi-lbl I have black on the most updated version of this one file on my computer; it said no changes. I also confirmed I do not have any uncommitted or unpushed changes. Yet the black test is still failing.

For everyone: I tested this on Google Colab (hence several commits this morning.) I am not sure how to automate testing in Google Colab or if that is really even needed.

**Comment by andrewlee94** [comment]
@adowling2 `black` is very temperamental - you need to have the correct version of `black` installed as the formatting changes slightly from version to version.

**Comment by adowling2** [comment]
@andrewlee94 Now that I have your attention, do you want to review this quickly?

**Review by andrewlee94 — APPROVED** [review]
This looks OK to me, but I am no expert on this sort of stuff.

File: scripts/colab_helper.py
Comment (jsiirola) [inline scripts/colab_helper.py:120]: This is a minor nit, but we probably want to put `idaes/bin` first and not last in the path (the idea being that we would be overriding anything that was previously installed into the system)

File: scripts/colab_helper.py
Comment (jsiirola) [inline scripts/colab_helper.py:49]: (More of an FYI than an action) This is find for collab (where we now it is Linux-based).  However, a more portable approach would be to use something akin to Pyomo's `pyomo.common.fileutils.Executable` (which is what Pyomo uses for locating executables)

File: scripts/colab_helper.py
Comment (jsiirola) [inline scripts/colab_helper.py:86]: If you are just going to print out the output anyway, why bother with `capture=True`?

**Review by jsiirola — COMMENTED** [review]
This is getting pretty close.

File: scripts/colab_helper.py
Comment (adowling2) [reply scripts/colab_helper.py]: I fixed this and tested it works on Colab.

File: scripts/colab_helper.py
Comment (jsiirola) [reply scripts/colab_helper.py:86]: On the terminal, the output comes to the terminal if you don't specify `capture_output`.  I wonder if this is different for the collab environment?  If that's the case, we want to at least be consistent and capture/print for *all* subprocess calls.

File: scripts/colab_helper.py
Comment (lbianchi-lbl) [inline scripts/colab_helper.py:37]: I'd recommend to add some versioning information, e.g. by adding a line here with `__version__ = "0.23.8"` (or any other date-based versioning schema).

**Review by lbianchi-lbl — CHANGES_REQUESTED** [review]
I think having this script here can be valuable for users of IDAES on Google Colab in addition to and beyond the original audience. A few general thoughts:

- I don't think there's a way to test this automatically (without an unreasonable amount of effort)
- Since the script is tested and will continue to be used in the future, I think we can pragmatically rely on this "outside testing" to detect possible breaking changes
- However, since we're adding code to the repository that's not being tested by the CI, before merging, we should:
  -  Identify who the maintainer(s) of this script will be, e.g. by adding an entry to the `CODEOWNERS` file
  - Add some kind of simple versioning information to the script itself (since the script won't have access to either the Git history or the `idaes` version at runtime)

File: scripts/colab_helper.py
Comment (adowling2) [reply scripts/colab_helper.py:86]: I expected that behavior on Colab, but no luck when testing in a cell.

Comment (adowling2) [reply scripts/colab_helper.py:99]: @jsiirola, I like your suggestion. I added a verbose flag to display the captured pip, idaes get-extensions, etc. output

File: scripts/colab_helper.py
Comment (adowling2) [reply scripts/colab_helper.py:37]: Great suggestion

**Comment by adowling2** [comment]
@jsiirola @lbianchi-lbl Thanks for the suggested changes. This should be ready.

Changes:
- Capture all subprocess output. Added verbose flag to toggle on output for pip install, idaes get-extensions, etc.
- Added a version
- Added my name in `CODEOWNERS`

**Comment by adowling2** [comment]
@jsiirola @lbianchi-lbl This is ready for your re-review.

**Review by jsiirola — APPROVED** [review]

**Pull request merged**
